internal/kubernetes/client.go

```diff
@@ -36,7 +40,13 @@ type Client interface {
 }
 
 type kubernetesClient struct {
-	client client.Client
+	client dynamic.Interface
+}
+
+var kubeconfig string
```

Comment:
It would be great if we didn't have to introduce a global state like this. I'm surprised that github.com/kubernetes-sigs/controller-runtime also takes this approach!